libraries/chain/controller.cpp

```diff
@@ -1497,9 +1497,12 @@ struct controller_impl {
          assert(!irreversible_mode() || forkdb.head());
          const auto& head_id = irreversible_mode() ? forkdb.head()->id() : chain_head.id();
          // verifies lib is on head branch, otherwise returns an empty branch
-         // The new lib needs to be on the head branch because the forkdb.advance_root() below could purge blocks that
-         // would be needed to be re-applied on a fork switch from the exiting chain_head.
-         auto branch = forkdb.fetch_branch(head_id, new_lib_id);
+         //   The new lib needs to be on the head branch because the forkdb.advance_root() below could purge blocks that
+         //   would be needed to be re-applied on a fork switch from the exiting chain_head.
+         // Pending LIB can be greater than chain head, for example when syncing, in that case fetch branch from the
+         // pending LIB. If the pending LIB not found on the head branch then fetch_branch returns an empty branch.
+         // Otherwise fetch_branch will return from chain_head to root iff chain_head on pending LIB branch.
+         auto branch = new_lib_num <= chain_head.block_num() ? forkdb.fetch_branch(head_id, new_lib_id) : forkdb.fetch_branch(new_lib_id, head_id);
```

Comment:
What if out current `head` is not an ancestor of `new_lib_id`. Should we do a fork switch then?

Reply:
Yes. And similar to the case where `new_lib_id` is not a decedent of `head` we will fork switch on the the next call to `maybe_switch_forks`.